### ISSUE TITLE
Media: Cannot view self-hosted images if traffic from Photon (i0.wp.com) is blocked

### ISSUE THREAD

**Issue by eeturunen** [wordpress-mobile/WordPress-iOS#14514]
### Expected behavior
Media is loaded directly from the WP host if Photon (WP CDN) returns an error

### Actual behavior
Cannot view media at all if Photon IP range is blocked

### Steps to reproduce the behavior
1. Block traffic from the IP range 195.234.109.0/24 to your WP host
2. Open the WP iOS app and view media either in the media library or within a blog post
3. Media does not load

This happens on a **fresh WP install without Jetpack or any plugins installed**.

Why does the iOS app insist on loading the images through Photon? And why doesn't it at least fall back to loading the images directly from the server if Photon returns an error?

In the logs it says "Unable to download image for attachment with url = Optional(https://i0.wp.com/...)

##### Tested on iOS 13.6, WPiOS 15.2

**Comment by rachelmcr** [comment]
@bummytime Looping you in because I'm not sure of the history here.

**Comment by bummytime** [comment]
@eeturunen thanks for the issue report. @leandroalonso any thoughts on this one?

**Comment by leandroalonso** [comment]
@SergioEstevao pinging you here because I know that you have more context than me when it comes to Media. :)

@bummytime I'm really not sure of the effort to fix that one. Unfortunately, we have different ways of loading media in different places, so I'm not sure about a straightforward solution. But I'm hoping @SergioEstevao can help me here.

**Comment by SergioEstevao** [comment]
@leandroalonso, you are correct there are multiple paths depending on if it's the editor, the reader, or the app table views.

- Editor: https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift#L125
- Table Views: (Post Lists, Sites Lists) https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Utility/WPTableImageSource.m#L137
- https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Utility/Media/ImageLoader.swift#L155

One good way to find photon uses is to search for references to `PhotonImageURLHelper` class.

But looking at all of them it looks that we default to Photon to any site that is public (without any password protection) WP.com or not.

The idea behind using photon was because of the speed/performance advantages.

We can change that code and only use Photon for WP.com and Jetpack sites and use direct access to all other WP sites.

We could also take advantage of this oportunity and try to have all this code paths be refactored to a single class, I think @diegoreymendez had a look on this recently.

**Comment by diegoreymendez** [comment]
I unified the authentication code for image downloads through the class `MediaRequestAuthenticator` as part of Private Atomic Sites.  I considered unifying the download code, but it went a bit too far beyond scope of what I was doing at the time.

But I digress...

I'm interested in understanding the background of the issue @eeturunen behind this report.  I agree that we should attempt a direct download if Photon is blocked.

But is there a reason / use case you have identified for blocking Photon beyond testing that the fallback works?

**Comment by eeturunen** [comment]
Thanks for the replies!

The use case is an intranet blog running on a server which only allows connections from within the organization's network.

**Comment by eeturunen** [comment]
Any progress on this?

**Comment by designsimply** [comment]
@eeturunen, out of curiosity, may I ask why it wouldn't be possible for you to allow traffic for 195.234.109.0/24 through on the network for this use case, in order to take advantage of the Photon CDN? Without that, it seems to me that it could cause really slow loading and increased data usage for users of the app in some cases, and not just for your own site but also for users elsewhere. (Please someone correct me if that's incorrect.)

**Comment by eeturunen** [comment]
@designsimply There is no need for a CDN in LAN.

**Comment by designsimply** [comment]
My thinking was that others who are not on a LAN would also be affected by the change you proposed to not use Photon (which I think we would want to avoid), but maybe I'm not 100% clear on the proposed change and how it would affect others.

If the proposed change is to attempt a direct download if Photon is blocked, I agree with that for sure. 👍

**Comment by eeturunen** [comment]
Exactly. There should at least be fallback to direct dl if an image cannot be retrieved through Photon.

Nevertheless, I think users should be given an option to decide whether to utilize Photon or not.

**Comment by eeturunen** [comment]
I think this issue is still relevant. Images should not be served from Photon servers without the administrator's consent. It might be a GDPR issue as well.

**Comment by leandroalonso** [comment]
Hi @eeturunen!

Thanks for you feedback, and we’ll certainly consider making use of our CDN optional. However, we don’t see this as a GDPR or privacy concern; our CDN service is part of our mobile app in order to provide the best speed and performance.

**Comment by holyspidoo** [comment]
This completely breaks the app for self hosted wordpress (which only your family accesses for example). I don't understand why we can't have a fallback for the CDN url or disable option, seems easy enough?
